### ISSUE TITLE
Change log: wrong changes recorded (de/en mismatch)

### ISSUE THREAD

**Issue by UlrichKu** [chameleon-system/chameleon-system#269]
Followup of #228.

**Describe the bug**
The problem of #228 still occurs partly.
For example if the current (Profile) display language is German and the current edit language is English.

**Affected version(s)**
6.2.6

**Technical details**
Problem manifests in \TCMSTableConf::getDataForCurrentLanguage().
Here the "language" is wrong in the above case.
And interestingly it is only wrong in the "change log" case - there are other calls to this method which yield the correct language.

**Comment by esono-anja** [comment]
Example: change the field "name" for a product with variants

Various errors occur in the changelog:
1. If you change a record, the Description field is always displayed with the changed values, although nothing has been changed here. Only if you change the same record again, only the actually changed value is displayed.
2. In the changelog overview, the variants are also always listed, although nothing has been changed here.

=> Affects all language combinations of editing and profile language.

further error:
3. in english editing language the old value is displayed in german => this should be displayed in english.

**Comment by UlrichKu** [comment]
Probably (only) the wrong thing deployed.

**Comment by esono-anja** [comment]
> 3. in english editing language the old value is displayed in german => this should be displayed in english.
=> this is solved.

Erros 1. an d2. still occur

**Comment by umulmrum** [comment]
Some findings:
- This happens independent of edit and display language as @esono-anja pointed out.
- This happens only once for a product. If we ever save the product again, it works as expected.
- The problem lies within the WYSIWYG fields. Only those fields are saved in the changelog although they are not unchanged.
- When looking into the details of the changes, the serialized string is shown with another length after change. I couldn't see any changes in the string, though.
- I suppose that problem 2 is solved automatically if problem 1 is solved, as the variant descriptions depend on the main product description.

One possible solution I can imagine is that ckEditor saves different line endings depending on the OS - at least this would explain "invisible" line changes, as Windows uses \r\n and Linux uses \n only.

Could you have another look @UlrichKu ?

**Comment by UlrichKu** [comment]
> One possible solution I can imagine is that ckEditor saves different line endings depending on the OS - at least this would explain "invisible" line changes, as Windows uses \r\n and Linux uses \n only.

Something like this seems to be the case.
I looked at the descriptions of some articles and they either end with \n or \r\n.
That explains especially why such a change log entry only occurs once per user/test.

**Comment by UlrichKu** [comment]
> > One possible solution I can imagine is that ckEditor saves different line endings depending on the OS - at least this would explain "invisible" line changes, as Windows uses \r\n and Linux uses \n only.
> 
> Something like this seems to be the case.
> I looked at the descriptions of some articles and they either end with \n or \r\n.
> That explains especially why such a change log entry only occurs once per user/test.

However configuring ckeditor doesn't seem to help.
At least not with protectedSource: [ "/\r\n/g" ]

**Comment by umulmrum** [comment]
OK, I moved this topic to another issue: chameleon-system/chameleon-system#379

The original problem here seems to be solved.
